src/instant-mocha.ts

```diff
@@ -25,8 +25,12 @@ export default async function instantMocha(
 
 	let webpackConfig: webpack.Configuration;
 	try {
-		// eslint-disable-next-line node/global-require
-		webpackConfig = require(webpackConfigPath);
+		// eslint-disable-next-line node/global-require, @typescript-eslint/no-var-requires
+		const config = require(webpackConfigPath);
+		webpackConfig = typeof config === 'function'
+			// first parameter is the -env flag on webpack, second is argv.
+			? config({}, { mode: options.mode })
+			: config;
```

Comment:
Mode should get overwritten here too

Reply:
Are you saying it watch is present it should force the mode to be development? I think the code in the documentation above is just an example on how you could use the environment variable to change the mode if desired. I would say 99% of the time if --watch is used then mode is likely development also but I would imagine if watch and mode present then to keep the mode value. If only watch is present then mode can be set to development.

Not even sure I can come up with an use case for --watch and --mode = production, but I tend to lean on the side of trusting the user knows what they are doing.

Reply:
When using webpack-cli with `--mode`, it seems to overwrite the mode in the config file.

Reply:
As in if the config is not a function you want me to set config.mode = options.mode if mode exists?

Reply:
Yeah exactly